src/Avalonia.Base/Media/RenderOptions.cs

```diff
@@ -1,36 +1,104 @@
         /// <returns>The control's left coordinate.</returns>
-        public static BitmapInterpolationMode GetBitmapInterpolationMode(AvaloniaObject element)
+        public static BitmapInterpolationMode GetBitmapInterpolationMode(Visual element)
         {
-            return element.GetValue(BitmapInterpolationModeProperty);
+            return element.RenderOptions.BitmapInterpolationMode;
         }
 
         /// <summary>
         /// Sets the value of the BitmapInterpolationMode attached property for a control.
         /// </summary>
         /// <param name="element">The control.</param>
         /// <param name="value">The left value.</param>
-        public static void SetBitmapInterpolationMode(AvaloniaObject element, BitmapInterpolationMode value)
+        public static void SetBitmapInterpolationMode(Visual element, BitmapInterpolationMode value)
+        {
+            element.RenderOptions = element.RenderOptions with { BitmapInterpolationMode = value };
+        }
+
+        /// <summary>
+        /// Gets the value of the EdgeMode attached property for a control.
+        /// </summary>
+        /// <param name="element">The control.</param>
+        /// <returns>The control's left coordinate.</returns>
+        public static EdgeMode GetEdgeMode(Visual element)
+        {
+            return element.RenderOptions.EdgeMode;
+        }
+
+        /// <summary>
+        /// Sets the value of the EdgeMode attached property for a control.
+        /// </summary>
+        /// <param name="element">The control.</param>
+        /// <param name="value">The left value.</param>
+        public static void SetEdgeMode(Visual element, EdgeMode value)
+        {
+            element.RenderOptions = element.RenderOptions with { EdgeMode = value };
+        }
+
+        /// <summary>
+        /// Gets the value of the TextRenderingMode attached property for a control.
+        /// </summary>
+        /// <param name="element">The control.</param>
+        /// <returns>The control's left coordinate.</returns>
+        public static TextRenderingMode GetTextRenderingMode(Visual element)
+        {
+            return element.RenderOptions.TextRenderingMode;
+        }
+
+        /// <summary>
+        /// Sets the value of the TextRenderingMode attached property for a control.
+        /// </summary>
+        /// <param name="element">The control.</param>
+        /// <param name="value">The left value.</param>
+        public static void SetTextRenderingMode(Visual element, TextRenderingMode value)
         {
-            element.SetValue(BitmapInterpolationModeProperty, value);
+            element.RenderOptions = element.RenderOptions with { TextRenderingMode = value };
+        }
+
+        public RenderOptions MergeWith(RenderOptions other)
+        {
+            var bitmapInterpolationMode = BitmapInterpolationMode;
+
+            if (bitmapInterpolationMode == BitmapInterpolationMode.Default)
```

Comment:
I'd suggest to add `Unspecified` members to those enums, so we can make better decisions in this method. Right now it's not possible to downgrade interpolation mode to `Default`, since it will be force-overwritten